lib/sharing/index.ts

```diff
@@ -0,0 +1,307 @@
+  include?: string;
+}
+
+/**
+ * Options for listing shares
+ */
+export interface ListSharesOptions extends RequestOptions<ListSharesQueryParams, undefined> {
+  assetType: AssetType;
+  assetId: string | number;
+  maxItems?: number;
+  lastKey?: string;
+  sharingInclude?: string;
+}
+
+/**
+ * Options for getting a share
+ */
+export interface GetShareOptions extends RequestOptions<object, undefined> {
+  assetType: AssetType;
+  assetId: string | number;
+  shareId: string;
+}
+
+/**
+ * Options for sharing an asset
+ */
+export interface ShareAssetOptions extends RequestOptions<{ sendEmail?: boolean }, CreateShareRequest[]> {
+  assetType: AssetType;
+  assetId: string | number;
+  sendEmail?: boolean;
+  body: CreateShareRequest[];
+}
+
+/**
+ * Options for updating a share
+ */
+export interface UpdateShareOptions extends RequestOptions<object, UpdateShareRequest> {
+  assetType: AssetType;
+  assetId: string | number;
+  shareId: string;
+  body: UpdateShareRequest;
+}
+
+/**
+ * Options for deleting a share
+ */
+export interface DeleteShareOptions extends RequestOptions<object, undefined> {
+  assetType: AssetType;
+  assetId: string | number;
+  shareId: string;
+}
+
+/**
+ * Sharing API interface
+ */
+export interface SharingApi {
+  listAssetShares: (
+    options: ListSharesOptions,
+    callback?: RequestCallback<SharesResponse>
+  ) => Promise<SharesResponse>;
+  
+  getAssetShare: (
+    options: GetShareOptions,
+    callback?: RequestCallback<ShareResponse>
+  ) => Promise<ShareResponse>;
+  
+  shareAsset: (
+    options: ShareAssetOptions,
+    callback?: RequestCallback<SharesSuccessResult>
+  ) => Promise<SharesSuccessResult>;
+  
+  updateShare: (
+    options: UpdateShareOptions,
+    callback?: RequestCallback<ShareResponse>
+  ) => Promise<ShareResponse>;
+  
+  deleteShare: (
+    options: DeleteShareOptions,
+    callback?: RequestCallback<any>
+  ) => Promise<any>;
+}
+
+/**
+ * Creates and returns the sharing API
+ * @param options Create options
+ * @returns Sharing API
+ */
+export const createSharing = (options: CreateOptions): SharingApi => {
+  const requestor = options.requestor;
+  
+  const optionsToSend = {
+    ...options.clientOptions
+  };
+  
+  // Base URL for shares endpoints
+  const baseUrl = '/shares';
+
+  /**
+   * List all shares for a specified asset
+   * @param options Options for listing shares
+   * @param callback Optional callback
+   * @returns Promise with shares response
+   */
+  const listAssetShares = (
+    options: ListSharesOptions,
+    callback?: RequestCallback<SharesResponse>
+  ): Promise<SharesResponse> => {
+    const { assetType, assetId, maxItems, lastKey, sharingInclude } = options;
+
+    // Build the base URL with required parameters
+    const urlParams = new URLSearchParams();
+    urlParams.append('assetType', assetType.toString());
+    urlParams.append('assetId', assetId.toString());
+
+    // Add optional query parameters
+    if (maxItems !== undefined) {
```

Comment:
this then would become

```suggestion
    if (queryParameters.maxItems !== undefined) {
```